api/callbacks.go

```diff
@@ -46,22 +70,37 @@ func buildDB(kv KVStore) C.DB {
 }
 
 //export cGet
-func cGet(ptr *C.db_t, key C.Buffer, val C.Buffer) i64 {
+func cGet(ptr *C.db_t, key C.Buffer, val *C.Buffer) (ret C.GoResult) {
+	defer recoverPanic(&ret)
+	if val == nil {
+		// we received an invalid pointer
+		return C.GoResult_BadArgument
+	}
+
 	kv := *(*KVStore)(unsafe.Pointer(ptr))
 	k := receiveSlice(key)
 	v := kv.Get(k)
-	if len(v) == 0 {
-		return 0
+	// v will equal nil when the key is missing
+	// https://github.com/cosmos/cosmos-sdk/blob/1083fa948e347135861f88e07ec76b0314296832/store/types/store.go#L174
+	if v != nil {
```

Comment:
Yeah, i like Adding rationale in code comments, and part of that is linking to relevant snippets of foreign code :) In this case it also adds context because it is otherwise opaque which storage is actually being used here